src/esp/gfx/TextRenderer.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <Corrade/Containers/Containers.h>
+#include <Corrade/PluginManager/Manager.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/Math/Matrix3.h>
+#include <Magnum/Shaders/Vector.h>
+#include <Magnum/Text/AbstractFont.h>
+#include <Magnum/Text/Alignment.h>
+#include <Magnum/Text/GlyphCache.h>
+#include <Magnum/Text/Renderer.h>
+#include <string>
+#include <vector>
+
+namespace esp {
+namespace gfx {
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+
+const std::string defaultCharacters =
+    "abcdefghijklmnopqrstuvwxyz"
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    "0123456789?!:;,.# ";
+
+class TextRenderer {
+ public:
+  /**
+   * @brief Constructor
+   */
+  // aspectRatio = width / height
+  explicit TextRenderer(float viewportAspectRatio = 1.33333f,
+                        float fontSize = 20.0f,
+                        int cacheSize = 512,
+                        const std::string& characters = defaultCharacters);
+
+  /**
+   * @brief Create a renderer to render text on screen
+   * @return renderer ID
+   */
+  size_t createRenderer(
+      float onScreenCharacterSize = 0.08f,
```

Comment:
I would make the characters a bit smaller -- `0.04f`? (but keeping the glyph cache texture size a bit higher, as supersampling will make the text look crisper).